ccswbs/gus
# Refactor Content Hub index page

# Summary of changes
Improve the layout and structure of `index.js` so the list of content is better organized and easier to parse

## Frontend
- Upgrade packages to latest versions
- Add tag sorting to `index.js`
- Categorize basic pages by how they're tagged (currently pulling from the `Tags` and `Units` vocabularies)
- Include term descriptions if they exist
- Update intro text on `index.js` to be more informative
- Add extra column classes based on `two-col-md` to `global.css` (no Bootstrap 5 equivalent exists)

## Backend
- N/A

# Test Plan
- Go to https://tqtest.gatsbyjs.io/
- Verify basic pages are listed by tag and that term descriptions appear
- Verify multi-column layout switches to single column on narrow viewports
- Go to https://build-2cf8bdef-5dac-4ae5-9722-08b9fad77f86.gtsb.io/ 
- Verify `Unpublished Content` section is empty (because you're viewing a PR build that authenticates with the `gatsbyview` role which lacks permission to view unpublished content)

## Known issues
- Tag sorting is only implemented for basic pages at this time
- The `Basic Page` and `Programs` headings still appear under `Unpublished Content` even when no unpublished content exists

### Bug...or Feature?
- If a page has more than one tag, it will listed more than once
- The heading for untagged pages remains red so it stands out vs the others
